power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -779,23 +779,48 @@ class Deserializer {
         if (msg_data.size == 0) {
             return;
         }
+
         // set offset and skip array header
         offset_ = msg_data.offset;
         parse_map_array<visit_array_t, move_forward>();
+
         for (element_number_ = 0; element_number_ != msg_data.size; ++element_number_) {
             BufferView const element_buffer = advance(buffer_view, element_number_);
             // check the element is map or array
-            auto const element_visitor = parse_map_array<visit_map_array_t, move_forward>();
-            if (element_visitor.is_map) {
-                parse_map_element(row_or_column_tag, element_buffer, element_visitor.size, component);
-            } else {
-                parse_array_element(row_or_column_tag, element_buffer, element_visitor.size, component, attributes);
-            }
+            parse_element(row_or_column_tag, element_buffer, component, attributes);
         }
         element_number_ = -1;
         offset_ = 0;
     }
 
+    void parse_element(row_based_t tag, BufferView const& buffer_view, MetaComponent const& component,
+                       std::span<MetaAttribute const* const> attributes) {
+        assert(is_row_based(buffer_view));
+
+        auto const element_visitor = parse_map_array<visit_map_array_t, move_forward>();
+        if (element_visitor.is_map) {
+            parse_map_element(tag, buffer_view, element_visitor.size, component);
+        } else {
+            parse_array_element(tag, buffer_view, element_visitor.size, component, attributes);
+        }
+    }
+
+    void parse_element(columnar_t tag, BufferView const& buffer_view, MetaComponent const& component,
+                       std::span<MetaAttribute const* const> attributes) {
+        assert(is_columnar(buffer_view));
+
+        auto const element_visitor = parse_map_array<visit_map_array_t, stay_offset>();
+        if (element_visitor.is_map) {
+            parse_map_array<visit_map_array_t, move_forward>();
+            parse_map_element(tag, buffer_view, element_visitor.size, component);
+        } else if (!buffer_view.reordered_attribute_buffers.empty()) {
+            parse_map_array<visit_map_array_t, move_forward>();
+            parse_array_element(tag, buffer_view, element_visitor.size, component, attributes);
+        } else {
+            parse_skip();
```

Comment:
the peeking pattern enables to `skip` components that are specified using the array serialization schema, but do not have any attribute buffers specified that are deserialized